lib/class-sp-integration.php

```diff
@@ -108,10 +108,28 @@ public function remove_hooks() {
 	 * @return array|null
 	 */
 	public function filter__posts_pre_query( $posts, $query ) {
-		if ( ! $query->is_main_query() || ! $query->is_search() ) {
+		$should_auto_integrate = $query->is_main_query() && $query->is_search();
+		if (
+			/**
+			 * Should a query be automatically integrated with SearchPress?
+			 *
+			 * @param bool     $should_auto_integrate Whether the query should be auto-integrated with SearchPress. This
+			 * 									      defaults to true if the query is the main search query.
+			 * @param WP_Query $query                 The query object.
+			 */
+			! apply_filters( 'sp_integrate_query', $should_auto_integrate, $query )
+		) {
 			return $posts;
 		}
 
+		// If we put in a phony search term, remove it now.
+		if ( '1441f19754335ca4638bfdf1aea00c6d' === $query->get( 's' ) ) {
+			$query->set( 's', '' );
```

Comment:
An empty string for `s` [is sufficient to cause the query to be `is_search()`](https://github.com/WordPress/wordpress-develop/blob/0c7ddbd67a5ca8a2eacb0e034a08549fb190db28/src/wp-includes/class-wp-query.php#L882-L884), which means, I think, that if `s` was an empty string before, the logic that added the phony search term wouldn't have run:

https://github.com/alleyinteractive/searchpress/blob/14b34ee531b04efd119c109e4faff409319fc002/lib/class-sp-integration.php#L143-L149

With that in mind, is setting `s` to `null` more accurate here than an empty string?

Reply:
Thanks for calling attention to this, I gather you're right that there's some flawed logic here and we can probably set `s` to an empty string initially and then not have to return to it (or set it to `null`) later.

For the purpose of this PR, I'm going to leave it as-is, since this is fixing a regression, but I opened up #162 to investigate a better fix.